### PR TITLE
Feature/m2 401 import schedule popup

### PR DESCRIPTION
**Pull request by wildkiomi**
image
image
image
image
image
image

### CODE REVIEW

File: src/shared/components/FileUploader/FileUploader.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        startIcon={<StyledSvg width="18" height="18" id="export" />}
+      >
+        {t('downloadTemplate')}
+      </Button>
+      <Box>
+        {isLoading ? (
+          <StyledLinearProgress />
+        ) : (
+          <>
+            {file ? (
```
Comment (IhorNesterenko99) [inline src/shared/components/FileUploader/FileUploader.tsx:78]: could you pls avoid nested conditions?

File: src/modules/Dashboard/features/Applet/Schedule/ImportSchedulePopup/ImportSchedulePopup.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
+import { useEffect, useState } from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+
+import { FileUploader, Modal, SubmitBtnColor } from 'shared/components';
+import { ImportedFile } from 'shared/components/FileUploader/FileUploader.types';
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/Applet/Schedule/ImportSchedulePopup/ImportSchedulePopup.tsx:5]: pls make export FileUploader.types in index and move import of the ImportedFile to from 'shared/components'?

File: src/shared/components/FileUploader/FileUploader.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      (event as DragEvent)?.dataTransfer.files;
+    if (!files.length) {
+      return;
+    }
+    const file = files[0];
+    importTable(file)
+      .then((data) => {
+        setFile({ name: file.name, data });
+        onFileReady(file);
+        console.log(data);
```
Comment (kdanilenka32) [inline src/shared/components/FileUploader/FileUploader.tsx:51]: remove console.log(data);

File: src/modules/Dashboard/features/Applet/Schedule/ImportSchedulePopup/ImportSchedulePopup.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+import { ImportSchedulePopupProps, Steps } from './ImportSchedulePopup.types';
+
+export const ImportSchedulePopup = ({
+  isIndividual = false,
+  open,
+  onClose,
+}: ImportSchedulePopupProps) => {
+  const { t } = useTranslation('app');
+
+  const appletName = 'Pediatric Screener';
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/Applet/Schedule/ImportSchedulePopup/ImportSchedulePopup.tsx:18]: Pls use real data appletName
in Schedule.tsx
 const { result: appletData } = applets.useAppletData() ?? {};
<Legend legendEvents={preparedEvents} appletName={appletData?.displayName || ''} />
in Legend.tsx
export const Legend = ({ legendEvents, appletName }: LegendProps) => { ...
and pass appletName as a property

File: src/modules/Dashboard/features/Applet/Schedule/ImportSchedulePopup/ImportSchedulePopup.tsx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+  const respondentName = '0234 (John Doe)';
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/Applet/Schedule/ImportSchedulePopup/ImportSchedulePopup.tsx:19]: pls add TODO to replace respondent name with real data name

File: src/shared/components/FileUploader/FileUploader.tsx [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+      })
+      .catch(() => setError(invalidFileFormatError));
+
+    setTimeout(() => setIsLoading(false), 3000);
```
Comment (alex-i-e) [inline src/shared/components/FileUploader/FileUploader.tsx:55]: please move `setIsLoading(false)` into `.finally` of the promise above

File: src/shared/components/FileUploader/FileUploader.tsx [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+      return;
+    }
+    const file = files[0];
+    importTable(file)
+      .then((data) => {
+        setFile({ name: file.name, data });
+        onFileReady(file);
+      })
+      .catch(() => setError(invalidFileFormatError))
+      .finally(() => setTimeout(() => setIsLoading(false), 3000));
```
Comment (IhorNesterenko99) [inline src/shared/components/FileUploader/FileUploader.tsx:53]: Could you pls remove setTimeout?

File: src/shared/components/FileUploader/FileUploader.types.ts [hunk 8]
```diff
@@ -0,0 +1,10 @@
+export type ImportedFile = {
+  name: string;
+  data: Record<string, string | number>[];
+};
+
+export type FileUploaderProps = {
+  uploadLabel: JSX.Element;
+  onFileReady: (file: any) => void;
```
Comment (IhorNesterenko99) [inline src/shared/components/FileUploader/FileUploader.types.ts:8]: could you pls remove any, maybe replace with unknown

**Review by IhorNesterenko99 — APPROVED** [review]

**Review by muhurov — APPROVED** [review]

**Pull request merged**